### ISSUE TITLE
MacVim draws FZF windows slower than Terminal.app

### ISSUE THREAD

**Issue by tambourine-man** [macvim-dev/macvim#881]
When using FZF.vim, MacVim draws considerably slower than Terminal.app. 

It's particularly noticeable on large screens. 
Where Terminal seems to simply jump the cursor to the bottom of the screen, MacVim appears to move it line by line until it reaches the bottom.

Please check the GIF below
![MacVimDrawingSpeed](https://user-images.githubusercontent.com/7542560/55597352-864a5100-5723-11e9-9836-0d22be13f8df.gif)

**Comment by eirnym** [comment]
Do the changes in #858 fix this?

**Comment by tambourine-man** [comment]
Not really. To be clear, I downloaded the source from Github (as there doesn't seem to be a binary release yet), ran ./configure && make.
The final app bundle version is 8.1.1270 (155). Is that it?

**Comment by eirnym** [comment]
Have you downloaded the branch by s4y, or just master branch?

**Comment by eirnym** [comment]
in other words, could you check the latest commit message and add a comment to that PR? 

It has big performance boost in comparison to what we have in the master, but it's not perfect yet.

**Comment by eirnym** [comment]
Could you also put an exact test case to test performance difference without gaining a knowledge of FZF.vim?

**Comment by tambourine-man** [comment]
No, I downloaded and compiled from master. You mean this, right?
https://github.com/s4y/macvim/tree/stateful-render

Just tried and it seems just as slow. 

It doesn't appear at first to be a rendering issue, but instead, it's as if MacVim is moving the cursor down one line at a time as fast as it can, while Terminal Vim just places it at the bottom in one go. But I could be wrong, of course.

I haven't seen the issue outside FZF.vim, but I'll see if I can create a test case without it.

**Comment by sstepashka** [comment]
I have exactly the same issue...

**Comment by ychin** [comment]
What’s “fat”? Do you mean fzf? The issue was supposed to be fixed but I mostly left it open so I can close it when I add Metal rendering.

if the terminal version is slow too that could be an issue with the plug-in rather than MacVim itself. But it really depends if you meant fzf or another plugin. It would be useful if you have a short clip to show what the issue is.

**Comment by sstepashka** [comment]
@ychin yeah, I meant fzf. Sorry for the typo. The symptoms are the same as on the video for iterm2, alacrity, and GUI. I found a bug only for Mac version and I do not have a chance to test on a different platform...

**Comment by ychin** [comment]
Have you tried regular Vim? If it happens for iTerm and Alacritty rather than just GUI then this seems to me like a Vim / fzf rendering issue, and not specific to MacVim.
